### ISSUE TITLE
[Bug]: DELETE'ing 'persons' is unnecesarily slow. 

### ISSUE THREAD

**Issue by Deedasmi** [LemmyNet/lemmy#3339]
### Requirements

- [X] Is this a bug report? For questions or discussions use https://lemmy.ml/c/lemmy_support
- [X] Did you check to see if this issue already exists?
- [X] Is this only a single bug? Do not put multiple bugs in one issue.
- [X] Is this a backend issue? Use the [lemmy-ui](https://github.com/LemmyNet/lemmy-ui) repo for UI / frontend issues.

### Summary

Trying to clean up 20,000+ bot accounts that aren't real used. I'm effectively the only user on my instance. DELETEing persons from the database runs 'update site_aggregates' in a loop and takes an unnecessarily long time. Deleting even a single user takes about 14 seconds.

### Steps to Reproduce

1. Try to delete a user
2. Wait

### Technical Details

Running 

```
DELETE FROM public.person p WHERE p.id IN (SELECT id FROM public.person p WHERE p.local='true' AND p.admin='false');
```

Takes multiple hours, and I've head to break it down to smaller chunks by adding LIMIT 1000 as it will never finish. CTRL+c reveals that the DB is stuck on 

```
CONTEXT:  SQL statement "update site_aggregates sa
  set users = users - 1
  from site s
  where sa.site_id = s.id"
  ```
  
  This is a rather naive way of keeping counters. However, it still shouldn't take nearly 14 seconds on my admit idly shitty hardware.

### Version

BE 0.18.0

**Comment by RocketDerp** [comment]
>  I'm effectively the only user on my instance. 

My instance is the same. Do you have a lot of federation likes, comments, posts coming into your server? I do on mine, and the most frequent SQL statement running on my server is that table!  This query `SELECT "person"."id", "person"."name", "person"."display_name", "person"."avatar", "person"."banned", "person"."published", "person"."updated", "person"."actor_id", "person"."bio", "person"."local", "person"."private_key", "person"."public_key", "person"."last_refreshed_at", "person"."banner", "person"."deleted", "person"."inbox_url", "person"."shared_inbox_url", "person"."matrix_user_id", "person"."admin", "person"."bot_account", "person"."ban_expires", "person"."instance_id" FROM "person" WHERE (("person"."deleted" = $1) AND ("person"."actor_id" = $2)) LIMIT $3` is being run more than any other query in Lemmy. No interactive users, just peer federation servers. 

DELETE is doing writes to that table and locks may be causing the whole server to bind up.

**Comment by Deedasmi** [comment]
No, I have only a single community with no users on it. How do you tell what the top queries are?

**Comment by RocketDerp** [comment]
> How do you tell what the top queries are?

[I've been recommending that all Lemmy instances install pg_stat_statements extension on their server.](https://lemmy.ml/post/1361757) Lemmy is suffering from severe performance problems and scaling issues, and this is a basic tool to see how many times SQL statements are being executed and the average (mean) time of execution. It is suitable for use in a production system and doesn't require you to crank up a ton of logging activity on your server.

A rough, quick and dirty webapp was thrown together by me to run such SQL statements from a web browser: https://github.com/RocketDerp/lemmy_helper/tree/master - and there are other easy ways to run a SQL statement if you want to go another route.

**Comment by phiresky** [comment]
This should be fixed in 0.18.3. Please comment if it's still an issue
